src/esp/assets/ResourceManager.cpp

```diff
@@ -683,6 +721,113 @@ Magnum::Range3D ResourceManager::computeMeshBB(BaseMesh* meshDataGL) {
+  PTexMeshData& ptexMeshData = dynamic_cast<PTexMeshData&>(baseMesh);
+  const std::vector<PTexMeshData::MeshData>& submeshes = ptexMeshData.meshes();
+
+  for (uint32_t iEntry = 0; iEntry < absTransforms.size(); ++iEntry) {
+    // convert std::vector<vec3f> to std::vector<Mn::Vector3>
+    std::vector<Mn::Vector3> pos;
+    uint32_t meshID = staticDrawableInfo_[iEntry].meshID;
+    for (auto& p : submeshes[meshID].vbo) {
+      pos.emplace_back(p);
+    }
+
+    // transform the vertex positions to the world space
+    Mn::MeshTools::transformPointsInPlace(absTransforms[iEntry], pos);
+
+    // locate the scene node which contains the current drawable
+    scene::SceneNode& node = staticDrawableInfo_[iEntry].node;
+
+    // set the absolute axis aligned bounding box
+    node.setAbsoluteAABB(Mn::Range3D{Mn::Math::minmax<Mn::Vector3>(pos)});
+  }
+#endif
+}
+
+void ResourceManager::computeGeneralMeshAbsoluteAABBs() {
+  std::vector<Mn::Matrix4> absTransforms = computeAbsoluteTransformations();
+
+  CORRADE_ASSERT(absTransforms.size() == staticDrawableInfo_.size(),
+                 "ResourceManager::computeGeneralMeshAbsoluteAABBs: number of "
+                 "transforms does not match number of drawables.", );
+
+  for (uint32_t iEntry = 0; iEntry < absTransforms.size(); ++iEntry) {
+    uint32_t meshID = staticDrawableInfo_[iEntry].meshID;
+
+    Corrade::Containers::Optional<Magnum::Trade::MeshData3D>& meshData =
+        meshes_[meshID]->getMeshData();
+    CORRADE_ASSERT(meshData,
+                   "ResourceManager::computeGeneralMeshAbsoluteAABBs: the "
+                   "empty mesh data", );
+
+    // a vector to store the min, max pos for the aabb of every position array
+    std::vector<Mn::Vector3> bbPos;
+
+    // transform the vertex positions to the world space, compute the aabb for
+    // each position array
+    for (uint32_t jArray = 0; jArray < (*meshData).positionArrayCount();
+         ++jArray) {
+      std::vector<Mn::Vector3>& pos = (*meshData).positions(jArray);
+      std::vector<Mn::Vector3> absPos =
+          Mn::MeshTools::transformPoints(absTransforms[iEntry], pos);
+
+      std::pair<Mn::Vector3, Mn::Vector3> bb =
+          Mn::Math::minmax<Mn::Vector3>(absPos);
+      bbPos.push_back(bb.first);
+      bbPos.push_back(bb.second);
+    }
+
+    // locate the scene node which contains the current drawable
+    scene::SceneNode& node = staticDrawableInfo_[iEntry].node;
+
+    // set the absolute axis aligned bounding box
+    node.setAbsoluteAABB(Mn::Range3D{Mn::Math::minmax<Mn::Vector3>(bbPos)});
+
+  }  // iEntry
+}
+
+std::vector<Mn::Matrix4> ResourceManager::computeAbsoluteTransformations() {
+  // sanity check
+  if (staticDrawableInfo_.size() == 0) {
+    return std::vector<Mn::Matrix4>{};
+  }
+
+  // basic assumption is that all the drawables are in the same scene;
+  // so use the 1st element in the vector to obtain this scene
+  auto* scene = dynamic_cast<Mn::SceneGraph::Scene<
```

Comment:
dynamic_cast to a pointer will return nullptr on error instead of throwing an exception so you should check the pointer or use a dynamic_cast to reference.